ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -944,29 +953,29 @@ runDB standalone@DB{..} cmd =
                 (const $ pure ())
                 (map ApplyVal bs)
                 db
-    go hasFS Snap = do
-        -- TODO: Maybe @Flush@ should be its own command but that would require
-        -- the implementation to change so that 'ledgerDbOldest' doesn't require
-        -- a 'flush' to happen just before it is called.
-
-        -- TODO: This duplicates too much logic of the flushing mechanisms. The
-        -- right thing to do would be to factor out the functions and here just
-        -- call one function that does all the job.
-        (toFlush, bs, db') <- atomically $ do
+
+    go _ Flush = do
+        (toFlush, bs) <- atomically $ do
           (_, db) <- readTVar dbState
           bs <- readTVar dbBackingStore
           let (toFlush, db') = ledgerDbFlush DbChangelogFlushAllImmutable db
           modifyTVar dbState (\(rs, _) -> (rs, db'))
-          pure (toFlush, bs, db')
+          pure (toFlush, bs)
         flush bs toFlush
-
+        pure Flushed 
+    go hasFS Snap = do
+        _ <- go hasFS Flush
```

Comment:
```
-- TODO: Maybe @Flush@ should be its own command but that would require
-- the implementation to change so that 'ledgerDbOldest' doesn't require
-- a 'flush' to happen just before it is called.
```

Add some rewording of this comment back